OpenRA.Mods.Common/FindActors.cs

```diff
@@ -0,0 +1,99 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+
+namespace OpenRA.Mods.Common
+{
+	public static class FindActors
+	{
+		/// <summary>
+		/// Finds all the actors of which their health radius is intersected by a line (with a definable width) between two points.
+		/// </summary>
+		/// <param name="world">The engine world the line intersection is to be done in.</param>
+		/// <param name="lineStart">The position the line should start at</param>
+		/// <param name="lineEnd">The position the line should end at</param>
+		/// <param name="lineWidth">How close an actor's health radius needs to be to the line to be considered 'intersected' by the line</param>
+		/// <returns>A list of all the actors 'intersected' by the line</returns>
+		public static IEnumerable<Actor> FindActorsOnLine(this World world, WPos lineStart, WPos lineEnd, WDist lineWidth, WDist targetExtraSearchRadius)
+		{
+			// This line intersection check is done by first just finding all actors within a square that starts at the source, and ends at the target.
+			// Then we iterate over this list, and find all actors for which their health radius is at least within lineWidth of the line.
+			// For actors without a health radius, we simply check their center point.
+			// The square in which we select all actors must be large enough to encompass the entire line's width.
+			var xDir = Math.Sign(lineEnd.X - lineStart.X);
+			var yDir = Math.Sign(lineEnd.Y - lineStart.Y);
+
+			var dir = new WVec(xDir, yDir, 0);
+			var overselect = dir * (1024 + lineWidth.Length + targetExtraSearchRadius.Length);
+			var finalTarget = lineEnd + overselect;
+			var finalSource = lineStart - overselect;
+
+			var actorsInSquare = world.ActorMap.ActorsInBox(finalTarget, finalSource);
+			var intersectedActors = new List<Actor>();
+
+			foreach (var currActor in actorsInSquare)
+			{
+				long actorWidth = 0;
```

Comment:
That is fine.  Ints are automatically promoted to longs when you do an operation that involves both.